server/subscriptions_test.go

```diff
@@ -0,0 +1,95 @@
+package main
+
+import (
+	"encoding/json"
+	"github.com/mattermost/mattermost-server/v5/plugin/plugintest"
+	"sort"
+	"testing"
+)
+
+// pluginFromRepoNames returns mocked plugin for given repository names
+func pluginFromRepoNames(repoNames []string, chanelID string) *Plugin {
```

Comment:
Maybe we simplify this to
```suggestion
func pluginFromRepoNames(s subscriptions) *Plugin {
```

and let the caller build the `subscriptions`. This would add more flexibility to the caller.